### ISSUE TITLE
Inconsistency of solid/reactivity rule when dealing with event handlers?

### ISSUE THREAD

**Issue by pedro00dk** [solidjs-community/eslint-plugin-solid#168]
Hello, I have been using solid-js for more than two years and recently decided to use this tool.
I event recently made a small patch regarding one of the issues: https://github.com/solidjs-community/eslint-plugin-solid/pull/164

However, in my process of fixing linting issues I kept finding some cases that are inconsistent and IMHO could be handled differently. 
I made a list in the playground to identify all cases: https://playground.solidjs.com/anonymous/29d817a1-ae6e-419e-af52-01ccc5287064

![image](https://github.com/user-attachments/assets/a3687ea2-b431-4562-85b5-7e12d150d379)

Besides the issues from my patch, I found:

![image](https://github.com/user-attachments/assets/43b3347a-164d-478d-b3a4-35e0dc3972e7)
- These two should definitely trigger the solid/reactivity rule

<br/>
<br/>

![image](https://github.com/user-attachments/assets/ae28afd0-c774-4857-9ac4-e550b32e09de)
- This

**Comment by joshwilsonvu** [comment]
@pedro00dk Thanks again for your last patch and for doing the investigation here! These are some good points—I do think some regexes need to be adjusted to be as permissive as possible. 

- I'm with you that the async functions passed as props to `<Button>` should not warn.
- For
  ```jsx
  <button onclick={props.onClick} />, // WRONG - complain only about attribute casing
  ```
  I think one warning is enough. Though `onclick` is certainly intended to be an event handler, it could just as easily be something like `only` or `once`, hence the existing rule guiding the user to make their intention explicit. See #23
- For
  ```jsx
  <button on-click={props.onClick} />, // WRONG - handler is only attached once (even if custom event)
  ```
  I didn't even know the dash (-) syntax was legal! I'm happy for this to be treated like other event handlers.

I'd welcome a PR for these, but no obligation. I can take it as time allows if you don't get to it first. Thanks!

**Comment by ryansolid** [comment]
Triage on ≥0.16.0: several of the screenshotted cases have shifted — 0.16 split uncalled-accessor-in-attribute into `solid/no-accessor-as-prop`, graduated `event-handlers` in v2 mode, and #214/#215 fixed the returned-function and component-detection false positives. Could you re-run your playground list against 0.16 and note which inconsistencies survive? Happy to fix the remainder in the precision pass; otherwise we'll close as overtaken by 0.16.
